File: src/main/java/io/junseok/mealmate/domain/restaurant/dto/response/RestaurantInfo.java
package io.junseok.mealmate.domain.restaurant.dto.response;

import io.junseok.mealmate.domain.restaurant.entity.Restaurant;
import lombok.Builder;

@Builder
public record RestaurantInfo(
    String restaurantName,
    String restaurantImageUrl,
    String restaurantType,
    int likeCount) {

    public static RestaurantInfo fromEntity(Restaurant restaurant){
        return RestaurantInfo.builder()
            .restaurantName(restaurant.getRestaurantName())
            .restaurantImageUrl(restaurant.getRestaurantImageUrl())
            .restaurantType(restaurant.getRestaurantType())
            .likeCount(restaurant.getLikeCount())
            .build();
    }
}

File: src/main/java/io/junseok/mealmate/domain/restaurant/dto/response/RestaurantInfo.kt
package io.junseok.mealmate.domain.restaurant.dto.response;

import io.junseok.mealmate.domain.restaurant.entity.Restaurant


data class RestaurantInfo(
    val restaurantName: String,
    val restaurantImageUrl: String,
    val restaurantType: String,
    val likeCount: Int,
    val restaurantId: Long,
    val openAt: String,
    val closeAt: String,
)

fun Restaurant.toCreateRestaurantResponse() = this.restaurantId?.let {
    RestaurantInfo(
        restaurantName = this.restaurantName,
        restaurantImageUrl = this.restaurantImageUrl,
        restaurantType = this.restaurantType,
        likeCount = this.likeCount,
        restaurantId = it,
        openAt = this.openAt,
        closeAt = this.closeAt,
    )
}
